lib/auth/auth_with_roles.go

```diff
@@ -317,11 +330,73 @@ func (a *AuthWithRoles) UpsertNode(s services.Server) error {
 	return a.authServer.UpsertNode(s)
 }
 
+// filterNodes filters nodes based off the role of the logged in user.
+func (a *AuthWithRoles) filterNodes(nodes []services.Server) ([]services.Server, error) {
+	// For certain built-in roles, continue to allow access and return the full
+	// set of nodes to not break existing clusters during migration. This is
+	// also used by the proxy to cache a list of all nodes for it's smart
+	// resolution.
+	if a.hasBuiltinRole(string(teleport.RoleAdmin)) ||
```

Comment:
Why this code even necessary? If using standard built in roles should just allow to execute this action in the code above, so we have one less code path? Also, why RemoteProxy can list all remote nodes?

Reply:
never mind this comment, I understand that it's a special case now, although I don't get why would RemoteProxy has this permission still?

Reply:
For sure I know we have a test that uses `teleport.RoleRemoteProxy`. When I went to check where it was used with Trusted Clusters I found they were broken on master so I have not been able to check.

Will check this when Trusted Clusters are fixed.